File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/GradeByStudentToInstructorModel.kt
package com.example.drivingschoolappandroidclient.api.models

/**
 * Модель для передачи данных оценки на сервер
 */
open class GradeByStudentToInstructorModel(
    var classId: Int,
    var grade: Byte,
    var comment: String?,
){
    var gradeEnum : GradesByStudentsToInstructors
        get() = GradesByStudentsToInstructors.ofValue(grade)
        set(value) {
            grade = value.value
        }
}

File: PDP/DrivingSchoolWebAPI3/DrivingSchoolAPIModels/ApiModels/GradeByStudentToInstructorModel.cs
using System.ComponentModel.DataAnnotations;

namespace DrivingSchoolAPIModels
{
    /// <summary>
    /// Модель для передачи данных оценки на сервер
    /// </summary>
    public class GradeByStudentToInstructorModel
    {
        [Key]
        public int ClassId { get; set; }
        public GradesByStudentsToInstructors Grade { get; set; }
        public string? Comment { get; set; }
    }
}
